contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,262 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/DoubleEndedQueueUpgradeable.sol";
+
+import "./System.sol";
+
+interface IStakeHub {
+    function unbondPeriod() external view returns (uint256);
+    function transferGasLimit() external view returns (uint256);
+}
+
+contract StakePool is Initializable, ReentrancyGuardUpgradeable, ERC20Upgradeable, System {
```

Comment:
change `StakePool` to `StakeCredit`? Two reasons:
1. We mention `StakeCredit` in the BEP, it is a good reflection here.
2. This contract is a ERC20 token, so it is a token actually,  `Credit` seems better than `Pool`

Reply:
Done